wazo_call_logd/tests/test_generator.py

```diff
@@ -166,3 +170,63 @@ def _generate_cel_for_call(self, linked_id, cel_count=3):
             result.append(Mock(linkedid=linked_id))
 
         return result
+
+
+def mock_data_dict(**kwargs):
+    d = defaultdict(lambda: None)
+    d.update(**kwargs)
+    return d
+
+
+class TestParticipantProcessor(TestCase):
+    def setUp(self):
+        self.confd = Mock()
+        self.processor = _ParticipantsProcessor(self.confd)
+
+    def test_participants_missing_from_confd(self):
+        call_log = mock_call()
+        call_log.raw_participants = {
+            "channel/id": {
+            }
+        }
+        call_log.participants_info = [
+            {
+                "user_uuid": "some-user-uuid",
+                "answered": True
+            }
+        ]
+        self.confd.users.get.side_effect = requests.exceptions.HTTPError()
+        call_log = self.processor(call_log)
+        assert self.confd.mock_calls
+
+    def test_participant_identified_from_channel(self):
+        call_log = mock_call()
+        channel_name = "PJSIP/rgcZLNGE-00000028"
+        call_log.raw_participants[channel_name] = {
+            "role": "destination"
+        }
+        call_log.participants_info = [
+            {
+                "user_uuid": "some-user-uuid",
+                "answered": True,
+                "role": "destination"
+            }
+        ]
+
+        confd_user = mock_data_dict(uuid="some-user-uuid", lines=[mock_data_dict(name="rgcZLNGE", id=1)])
+        confd_line = mock_data_dict(name="rgcZLNGE", id=1, users=[confd_user])
+        self.confd.users.get.return_value = confd_user
+        self.confd.lines.list.return_value = {
+            "items":  [
+                confd_line
+            ]
+        }
+        call_log = self.processor(call_log)
+        self.assertEqual(len(self.confd.users.get.mock_calls), 1) # verify cache effectiveness
```

Comment:
You should avoid mixing different assertion methods in the same test (i.e. hamcrest vs unittest).
Also, I don't understand the comment about "cache effectiveness", are we supposed to have multiple calls without the cache, even with one participant only?

Reply:
Yes, because there is a `raw_participant` with a line corresponding to the same user in `participants_info`. Without a cache, `find_participant` would be called to find the user through the line, and then `find_participant_by_uuid` called on the `user_uuid` from `participants_info`.